addons/l10n_cz/models/account_move.py

```diff
@@ -5,7 +5,12 @@
 class AccountMove(models.Model):
     _inherit = 'account.move'
 
-    taxable_supply_date = fields.Date(default=fields.Date.today())
+    taxable_supply_date = fields.Date(compute='_compute_default_taxable_supply_date', store=True, readonly=False, precompute=True)
+
+    @api.depends('country_code')
+    def _compute_default_taxable_supply_date(self):
+        for move in self.filtered(lambda m: m.country_code == 'CZ' and not m.taxable_supply_date):
+            move.taxable_supply_date = fields.Date.today()
```

Comment:
@william-andre would that be better or does it make no difference?

```suggestion
            move.taxable_supply_date = fields.Date.context_today
```